native/src/test/java/io/xlibb/gateway/generator/common/CommonUtilTest.java

```diff
@@ -0,0 +1,171 @@
+
+import graphql.language.BooleanValue;
+import graphql.language.FloatValue;
+import graphql.language.IntValue;
+import graphql.language.StringValue;
+import graphql.language.Value;
+import graphql.schema.GraphQLFieldDefinition;
+import io.xlibb.gateway.GatewayProject;
+import io.xlibb.gateway.exception.GatewayGenerationException;
+import io.xlibb.gateway.exception.ValidationException;
+import io.xlibb.gateway.generator.GatewayCodeGenerator;
+import io.xlibb.gateway.generator.GatewayTestUtils;
+import io.xlibb.gateway.generator.GraphqlTest;
+import org.testng.Assert;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import java.io.File;
+import java.io.IOException;
+import java.math.BigDecimal;
+import java.math.BigInteger;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Objects;
+
+/**
+ * Class to test common utils used in graphql gateway generation.
+ */
+public class CommonUtilTest extends GraphqlTest {
+
+    @Test(description = "Test getting defined type objects", dataProvider = "SchemaAndTypeNamesProvider")
+    public void testGetCustomDefinedObjectTypeNames(GatewayProject project, String[] typeNames) {
+        Object[] namesFound = CommonUtils.getCustomDefinedObjectTypeNames(project.getSchema()).toArray();
+        Assert.assertEqualsNoOrder(typeNames, namesFound);
+    }
+
+    @DataProvider(name = "SchemaAndTypeNamesProvider")
+    public Object[][] getSchemaAndTypeNames() throws ValidationException, IOException {
+        return new Object[][]{
+                {GatewayTestUtils.getGatewayProject("two_entities", tmpDir),
+                        new String[]{"Astronaut", "Mission"}},
+                {GatewayTestUtils.getGatewayProject("two_entities_with_id_type_fields", tmpDir),
+                        new String[]{"Astronaut", "Mission"}},
+                {GatewayTestUtils.getGatewayProject("three_entities", tmpDir),
+                        new String[]{"Review", "Product", "Category"}},
+
+        };
+    }
+
+    @Test(description = "Test get query types", dataProvider = "SchemaAndQueryTypesProvider")
+    public void testGetQueryTypes(GatewayProject project, String[] queryTypes) {
+        Object[] queryTypeNames = CommonUtils.getQueryTypes(project.getSchema()).stream().map(
+                field -> ((GraphQLFieldDefinition) field).getName()
+        ).toArray();
+        Assert.assertEqualsNoOrder(queryTypeNames, queryTypes);
+    }
+
+    @DataProvider(name = "SchemaAndQueryTypesProvider")
+    public Object[][] getSchemaAndQueryTypes() throws ValidationException, IOException {
+        return new Object[][]{
+                {GatewayTestUtils.getGatewayProject("two_entities", tmpDir), new String[]{
+                        "astronaut", "astronauts", "mission", "missions", "serviceName", "isExist"
+                }}
+        };
+    }
+
+    @Test(description = "Test get mutation types", dataProvider = "SchemaAndMutationTypesProvider")
```

Comment:
I don't think these descriptions are adding any value. The test function name is self-explanatory.